src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -14,7 +15,7 @@ public class TelemetryActivity : IDisposable
         private readonly DateTime _startTime;
         private readonly Stopwatch _stopwatch;
         private readonly Stopwatch _intervalWatch = new Stopwatch();
-        private readonly List<Tuple<string, TimeSpan>> _intervalList;
+        private readonly ConcurrentBag<Tuple<string, TimeSpan>> _intervalList;
```

Comment:
Why do we need a concurrent bag here?
Is there a concurrency problem? Is this new?

Reply:
The current pattern of using the interval list is sequential
StartIntervalMeasure()
// do some work
EndIntervalMeasure()

With new proposed API 
``` csharp
 public void EndIntervalMeasure(string propertyName, Stopwatch stopWatch)
```

the list _intervalList could be updated concurrently.

Reply:
Can we use a simple object lock? 

It should be more than sufficient because a ConcurrentBag is really optimized for read/write scenarios.

Reply:
I think we can move https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Clients/NuGet.VisualStudio.Common/Telemetry/IntervalTracker.cs all the way to NuGet.Common because it's a scenario that we'll need as we try to reduce the number of events NuGet is sending. 

We might need to remove the Etw part of it though :) Hence why I called for @zivkan's opinion :)

Reply:
I think `IntervalTracker` only makes sense in code that is instrumented to emit telemetry.  If we go towards my goal of raising ETW events in the shared code, and listen to those ETW events in the VS code and transform it into a telemetry event, then there might not be a use for `IntervalTracker` in `NuGet.Common`. It would just be a wrapper for `Stopwatch`, without any real benefit, right?

Reply:
It allows you to have an easy "Start/Stop" pattern, where it's a disposable instead of having to create your own stopwatch to pass to the TelemetryActivity. 

I think it makes sense for tracking internals :) Whether you send telemetry is up to the user.